src/main.rs

```diff
@@ -1,120 +1,116 @@
 
     // Sort posts by date (newest first)
     site_data.posts.sort_by(|a, b| b.date.cmp(&a.date));
     // Sort pages on title
     site_data.pages.sort_by(|a, b| b.title.cmp(&a.title));
 
     // Create the output directory
-    let output_path = folder.join(&site_data.site.site_path);
+    let output_path = output_folder.join(&site_data.site.site_path);
     if let Err(e) = fs::create_dir_all(&output_path) {
         eprintln!("Unable to create output directory: {}", e);
         process::exit(1);
     }
 
     // Initialize Tera templates
-    let tera = match Tera::new(format!("{}/**/*", site_data.site.templates_path).as_str()) {
+    let templates_path = input_folder.join(&site_data.site.templates_path);
+    let tera = match Tera::new(&format!("{}/**/*.html", templates_path.display())) {
         Ok(t) => t,
         Err(e) => {
-            eprintln!("Parsing error(s): {}", e);
+            eprintln!("Parsing error(s) in templates: {}", e);
             process::exit(1);
         }
     };
+
     // Render templates
-    if let Err(e) = render_templates(&site_data, &tera, &folder.join(&site_data.site.site_path)) {
+    if let Err(e) = render_templates(&site_data, &tera, &output_path) {
         eprintln!("Failed to render templates: {}", e);
         process::exit(1);
     }
 
     // Copy static folder if present
-    let static_source = Path::new(site_data.site.static_path);
-    let static_destiny = output_path.join(Path::new("static"));
+    let static_source = input_folder.join(site_data.site.static_path);
+    let static_destiny = output_folder.join("static");
     if static_source.exists() {
         if let Err(e) = fs::create_dir_all(&static_destiny) {
             eprintln!("Unable to create static directory: {}", e);
             process::exit(1);
         }
-        for entry in WalkDir::new(&static_source) {
-            match entry {
-                Ok(entry) => {
-                    let static_filename = match entry.path().strip_prefix(&static_source) {
-                        Ok(filename) => filename,
-                        Err(e) => {
-                            eprintln!("Error building static_filename: {}", e);
-                            process::exit(1);
-                        }
-                    };
-                    let target_path = static_destiny.join(&static_filename);
-                    if entry.file_type().is_dir() {
-                        fs::create_dir_all(&target_path)?;
-                    } else {
-                        fs::copy(entry.path(), target_path)?;
-                    }
-                }
-                Err(e) => eprintln!("Error copying static file: {}", e),
-            }
+
+        let mut options = CopyOptions::new();
+        options.overwrite = true; // Overwrite files if they already exist
+
+        if let Err(e) = copy(&static_source, &static_destiny, &options) {
+            eprintln!("Failed to copy static directory: {}", e);
+            process::exit(1);
```

Comment:
In the refactored version, I used the `fs_extra` module with the `copy` function and the `CopyOptions`, which eliminates the need to manually check the file type and handle errors. Additionally, the `overwrite` option was enabled to allow replacing existing files.

This simplification reduces boilerplate code (I’m a Java developer "JAVEIRO", after all), makes the workflow more straightforward, and improves readability by encapsulating the file copy logic in a single function from an external library, instead of implementing all the logic manually.